src/actions/measurements.ts

```diff
@@ -517,14 +577,176 @@ export const changeMeasurementsDisplay = (
+    measurementsColorGrouping: groupingValues.includes(controls.measurementsColorGrouping)
+      ? controls.measurementsColorGrouping
+      : undefined,
+    measurementsGroupBy: newGroupBy
+  };
 
-  dispatch({
-    type: CHANGE_MEASUREMENTS_GROUP_BY,
-    controls: newControls,
-    queryParams: createMeasurementsQueryFromControls(newControls, measurements.collectionToDisplay, measurements.defaultCollectionKey)
+  batch(() => {
+    dispatch({
+      type: CHANGE_MEASUREMENTS_GROUP_BY,
+      controls: newControls,
+      queryParams: createMeasurementsQueryFromControls(newControls, measurements.collectionToDisplay, measurements.defaultCollectionKey)
+    });
+
+    /* After the group by has been updated, update the measurement coloring data if needed */
+    updateMeasurementsColorData(
+      newControls.measurementsColorGrouping,
+      controls.measurementsColorGrouping,
+      controls.colorBy,
+      controls.defaults.colorBy,
+      dispatch
+    );
+  })
+}
+
+export function getActiveMeasurementFilters(
+  filters: MeasurementFilters
+): {string?: string[]} {
+  // Find active filters to filter measurements
+  const activeFilters: {string?: string[]} = {};
+  Object.entries(filters).forEach(([field, valuesMap]) => {
+    activeFilters[field] = activeFilters[field] || [];
+    valuesMap.forEach(({active}, fieldValue) => {
+      // Save array of active values for the field filter
+      if (active) activeFilters[field].push(fieldValue);
+    });
+  });
+  return activeFilters;
+}
+
+export function matchesAllActiveFilters(
+  measurement: Measurement,
+  activeFilters: {string?: string[]}
+): boolean {
+  for (const [field, values] of Object.entries(activeFilters)) {
+    const measurementValue = measurement[field];
+    if (values.length > 0 &&
+       ((typeof measurementValue === "string") && !values.includes(measurementValue))){
+      return false;
+    }
+  }
+  return true;
+}
+
+function createMeasurementsColoringData(
+  filters: MeasurementFilters,
+  groupBy: string,
+  groupingValue: string,
+  collection: Collection,
+): {
+  nodeAttrs: MeasurementsNodeAttrs,
+  colorings: Colorings,
+} {
+  const measurementColorBy = encodeMeasurementColorBy(groupingValue);
+  const activeMeasurementFilters = getActiveMeasurementFilters(filters);
+  const strainMeasurementValues: {[strain: string]: number[]} = collection.measurements
+    .filter((m) => m[groupBy] === groupingValue && matchesAllActiveFilters(m, activeMeasurementFilters))
+    .reduce((accum, m) => {
+      (accum[m.strain] = accum[m.strain] || []).push(m.value)
+      return accum
+    }, {});
+
+  const nodeAttrs: MeasurementsNodeAttrs = {};
+  for (const [strain, measurements] of Object.entries(strainMeasurementValues)) {
+    const averageMeasurementValue = measurements.reduce((sum, value) => sum + value) / measurements.length;
+    nodeAttrs[strain] = {
+      [measurementColorBy]: {
+        value: averageMeasurementValue
+      }
+    };
+  }
```

Comment:
> Maybe when we’ve colored by measurements, we could have a filter generated that hides the tips without measurements by default but which we could remove to show those tips as needed?

@huddlej thanks for bringing up the need for filtering to measurements tips! I think this can be accomplished similar to how we add a [boolean field for the drag-and-drop metadata](https://github.com/nextstrain/auspice/blob/8a6e56fa4b926de7230bf3bd35fa9e590d878fd5/src/actions/filesDropped/metadata.js#L143-L150). Here, I can add a new field (e.g. `_measurementsColors`) here with the `true` value so that it can be used to filter to those tips.

Reply:
Added in 972ed91ef8cd1c173b969c3978ff66f0af1ca22e